gitnamehere/point-of-sale-group-project
# Customer Account Creation

Functionality for customers to create accounts on customer-facing website (`/store`)

Ticket: [ATS-56](https://cs375-pos-group-project.atlassian.net/jira/core/projects/ATS/board?selectedIssue=ATS-56)

- Created input fields for customer information:
   - First Name
   - Last Name
   - Phone Number
   - Email
- Created API endpoint for creating customer account `api/auth/store/account/create`
- Added validation for customer info fields, including handling duplicate emails

Testing:

- npm run start:dev
- go to `http://localhost:3000/store/account-management/`
- Ensure that site exists
- Create valid account
- Ensure that "Success" appears at top of page with 200 status code in Network tab
- Go into psql database
- Ensure that account has been added to the `customer` table with all relevant information
- Try to make another account with the same email used previously
- Ensure that "Email Already Exists" appears as failure message at top of page with 500 status code in Network tab
- Ensure that second account has not been added to psql database
